app/src/main/kotlin/com/simplemobiletools/clock/extensions/Activity.kt

```diff
@@ -0,0 +1,33 @@
+package com.simplemobiletools.clock.extensions
+
+import com.simplemobiletools.clock.BuildConfig
+import com.simplemobiletools.commons.activities.BaseSimpleActivity
+import com.simplemobiletools.commons.dialogs.PermissionRequiredDialog
+import com.simplemobiletools.commons.extensions.canUseFullScreenIntent
+import com.simplemobiletools.commons.extensions.openFullScreenIntentSettings
+
+fun BaseSimpleActivity.handleFullScreenNotificationsPermission(
+    fullScreenNotificationsDeniedCallback: () -> Unit = {},
+    notificationsCallback: (granted: Boolean) -> Unit,
```

Comment:
this looks pretty confusing, cant we merge 2 callbacks into 1 with granted parameter?

Reply:
Let me try to refactor it into something nicer. I needed a callback when users dimiss the dialog for full screen notification permission, to be able to update the view. Otherwise the app displays as if the alarm has been enabled, when it hasn't been.

Reply:
well, dismiss and decline should behave the same